lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
> I hope I don't come across as negative,

Me as well.  Thanks for bringing it up.  I think I was being a little snippy.

In terms of the feature, I'm not married to the inner workings, but I do like the `Adapter.get` and `Adapter.register` as the source of truth for getting and setting adapters. And that it removes the requirement that a custom adapter be in the ActiveModel::Serializer::Adapter namespace

In terms of borrowing the interface from ActiveJob, I think that's a really great idea!  cc @seuros